src/components/app/routes/data/utils.js

```diff
@@ -202,3 +204,100 @@ export function transformEnterpriseCustomer(enterpriseCustomer, enterpriseFeatur
     enterpriseFeatures,
   };
 }
+
+/**
+ * Transforms the redeemable policies data by attaching the subsidy expiration date
+ * to each assignment within the policies, if available.
+ * @param {object[]} [policies] - Array of policy objects containing learner assignments.
+ * @returns {object} - Returns modified policies data with subsidy expiration dates attached to assignments.
+ */
+export function transformRedeemablePoliciesData(policies = []) {
+  return policies.map((policy) => {
+    const assignmentsWithSubsidyExpiration = policy.learnerContentAssignments?.map(assignment => ({
+      ...assignment,
+      subsidyExpirationDate: policy.subsidyExpirationDate,
+    }));
+    return {
+      ...policy,
+      learnerContentAssignments: assignmentsWithSubsidyExpiration,
+    };
+  });
+}
+
+/**
+ * Takes a flattened array of assignments and returns an object containing
+ * lists of assignments for each assignment state.
+ *
+ * @param {Array} assignments - List of content assignments.
+ * @returns {{
+*  assignments: Array,
+*  hasAssignments: Boolean,
+*  allocatedAssignments: Array,
+*  hasAllocatedAssignments: Boolean,
+*  canceledAssignments: Array,
+*  hasCanceledAssignments: Boolean,
+*  acceptedAssignments: Array,
+*  hasAcceptedAssignments: Boolean,
+* }}
+*/
+export function getAssignmentsByState(assignments = []) {
```

Comment:
[inform] Copied from another file. The original defined function remains for now, but will eventually be removed once its no longer in use by the UI components yet to be refactored.